diff/windows/windows.go

```diff
@@ -191,3 +326,52 @@ func mountsToLayerAndParents(mounts []mount.Mount) (string, []string, error) {
 
 	return mnt.Source, parentLayerPaths, nil
 }
+
+// Ensure the two sets of mount-lists are actually a correct parent-and-child, or
+// orphan-and-empty-list, and return the full list of layers, starting with the
+// upper-most (most childish?)
+func mountPairToLayerStack(lower, upper []mount.Mount) ([]string, error) {
+
+	// May return an ErrNotImplemented, which will fall back to LCOW
+	upperLayer, upperParentLayerPaths, err := mountsToLayerAndParents(upper)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Upper mount invalid")
+	}
+
+	// Trivial case, diff-against-nothing
+	if len(lower) == 0 {
+		if len(upperParentLayerPaths) != 0 {
+			return nil, errors.Wrap(errdefs.ErrInvalidArgument, "windowsDiff cannot diff a layer with parents against a null layer")
+		}
+		return []string{upperLayer}, nil
+	}
+
+	if len(upperParentLayerPaths) < 1 {
+		return nil, errors.Wrap(errdefs.ErrInvalidArgument, "windowsDiff cannot diff a layer with no parents against another layer")
+	}
+
+	// May return an ErrNotImplemented, which will fall back to LCOW
```

Comment:
Is this still true? SHould we handle the error here?

Reply:
Good point. At this point, that suggests we've been asked to diff a windows-layer against a non-windows-layer. It makes sense to fail properly, as I don't think that's expected to be rational.